test/helpers/fixtures.js

```diff
@@ -32,7 +33,8 @@ async function initRouterContracts() {
     await tokens.DAI.approve(uniswapv2, ether('1'));
     await tokens.DAI.approve(uniswapv3, ether('1'));
     await tokens.DAI.approve(paraswap, ether('1'));
-    await tokens.DAI.approve(uniswapUniversalRouter, ether('1'));
+    await tokens.DAI.approve(permit2, ether('1'));
```

Comment:
`permit2` is not used anywhere, isn't it?

Reply:
Uniswap's universal router uses permit2 for all allowance handling. similar to matcha's latest contract. Approvals to the router will not be used so permit2 allowances is required from my understanding.

Reply:
In the example usage of the UniversalRouter that I provided above in the comment, `permit2` is not used, and if this line is commented out, the transaction still executes successfully.

It seems not entirely accurate to compare swaps using `approve` with those using `permit2`. However, the swap implementation using `permit2` that you have done is still useful. We just need to add an implementation of swaps on other protocols using `permit2` for comparison. Let's move these tests (which use `permit2`) to the separate section and then we will add tests for another protocols later?